server/bundler/async-loader.js

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable no-var */
+
+var esprima = require( 'esprima' ),
+	estraverse = require( 'estraverse' ),
+	escodegen = require( 'escodegen' ),
+	debug = console.log.bind( console );
+	//debug = require( 'debug' )( 'calypso:async-loader' );
+
+/*
+ * A cheaper heuristics to avoid parsing every file hit.
+ */
+function requiresAsyncLoad( content ) {
+	return -1 !== content.indexOf( 'components/async-load' );
+}
+
+function replaceAsyncLoadCalls( content, filePath ) {
+	var ast = esprima.parse( content ),
```

Comment:
So I was thinking about this a bit over the weekend, and it seemed to me that using Webpack would lead to this; i.e. the need to re-parse every required file. Doesn't this seem very... heavy? Is there any noticeable impact on bundle time?

The conclusion I was led to was that it might make more sense to use a Babel plugin to manipulate the output ([docs](https://github.com/thejameskyle/babel-handbook/blob/master/translations/en/plugin-handbook.md#manipulation)). I also thought it could be useful to have this available in addition to the prop variant, so maybe a callback form `requireAsync( 'path', callback )` which calls the callback immediately with the result of `require( 'path' )` if code splitting is not enabled. What do you think?